src/main/java/edu/cornell/kfs/module/purap/batch/dataaccess/impl/JaggaerUploadDaoJdbc.java

```diff
@@ -100,92 +108,103 @@ public List<JaggaerContractAddressUploadDto> findJaggaerContractAddress(JaggaerC
+                StringUtils.SPACE);
+        return chunk;
     }
     
-    private String buildRestrictActiveRows(boolean includeVendorAddress) {
-        StringBuilder sb = new StringBuilder("AND VD.DOBJ_MAINT_CD_ACTV_IND = 'Y' ");
+    private CuSqlChunk buildRestrictActiveRows(boolean includeVendorAddress) {
+        CuSqlChunk chunk = CuSqlChunk.of("AND VD.DOBJ_MAINT_CD_ACTV_IND = ", 
+                CuSqlChunk.forParameter(KFSConstants.ACTIVE_INDICATOR));
         if (includeVendorAddress) {
-            sb.append("AND VA.DOBJ_MAINT_CD_ACTV_IND = 'Y' ");
+            chunk.append(" AND VA.DOBJ_MAINT_CD_ACTV_IND = ", 
+                    CuSqlChunk.forParameter(KFSConstants.ACTIVE_INDICATOR));
         }
-        return sb.toString();
+        chunk.append(StringUtils.SPACE);
+        return chunk;
     }
     
-    private String buildTimeFrameRestrictionClause(JaggaerContractUploadProcessingMode processingMode, String processingDate) {
-        StringBuilder sb = new StringBuilder();
+    private CuSqlChunk buildTimeFrameRestrictionClause(JaggaerContractUploadProcessingMode processingMode, Date processingDate) {
         if (processingMode == JaggaerContractUploadProcessingMode.PO) {
-            sb.append("AND VH.VNDR_HDR_GNRTD_ID IN (").append(buildPurchaseOrderLimitSubQuery(processingDate)).append(") ");
+            CuSqlChunk chunk = CuSqlChunk.of("AND VH.VNDR_HDR_GNRTD_ID IN (",
+                    buildPurchaseOrderLimitSubQuery(processingDate),
+                    ") ");
+            return chunk;
         } else {
-            sb.append("AND VD.LAST_UPDT_TS > TO_DATE('").append(processingDate).append("', 'YYYY-MM-DD') ");
+            SqlParameterValue dateParameter = new SqlParameterValue(Types.DATE, processingDate);
+            CuSqlChunk chunk = CuSqlChunk.of("AND VD.LAST_UPDT_TS > ",
+                    new CuSqlChunk(dateParameter),
+                    StringUtils.SPACE);
+            return chunk;
         }
-        return sb.toString();
     }
     
-    private String buildPurchaseOrderLimitSubQuery(String processingDate) {
-        StringBuilder sb = new StringBuilder("SELECT VNDR_HDR_GNRTD_ID FROM (");
-        sb.append("SELECT VNDR_HDR_GNRTD_ID, COUNT(1) ");
-        sb.append("FROM KFS.PUR_PO_T ");
-        sb.append("WHERE PO_LST_TRNS_DT > TO_DATE('").append(processingDate).append("', 'YYYY-MM-DD') ");
-        sb.append("GROUP BY VNDR_HDR_GNRTD_ID ");
-        sb.append("HAVING COUNT(1) > 1)");
-        return sb.toString();
+    private CuSqlChunk buildPurchaseOrderLimitSubQuery(Date processingDate) {
+        SqlParameterValue dateParameter = new SqlParameterValue(Types.DATE, processingDate);
+        CuSqlChunk chunk = CuSqlChunk.of("SELECT VNDR_HDR_GNRTD_ID FROM (",
+                "SELECT VNDR_HDR_GNRTD_ID, COUNT(1) ",
+                "FROM KFS.PUR_PO_T ",
+                "WHERE PO_LST_TRNS_DT > ",
+                new CuSqlChunk(dateParameter),
```

Comment:
Another spot that could be simplified using the two-arg CuSqlChunk.forParameter() method.

Reply:
ok